plugins/logs/plugin.go

```diff
@@ -401,12 +401,46 @@ func (p *Plugin) Start(ctx context.Context) error {
 // Stop stops the plugin.
 func (p *Plugin) Stop(ctx context.Context) {
```

Comment:
I wonder if the passed ctx is the better interface for this: If there's a graceful shutdown period configured, `Stop` would be called with a context that has that timeout.

We'd have to double check what happens when you call `Stop(ctx)` with a a context that has a "0 duration" timeout, but what we do here seems like it's shadowing what ctx is for (or how it's commonly used, at least)... 🤔

Reply:
Hmm, yeah how to properly use `context` is still somewhat confusing to me, so I appreciate the advice. Do you think we should set call `WithTimeout` on the context and pass that from the plugin managers `Stop` method already? I considered something like that but would that not mean that if any plugin called `cancel` on that context it would mean all configured plugins would consider the deadline to be exceeded? Maybe that's OK - I don't really know what level of isolation we want the plugins to run with.

Reply:
> Do you think we should set call WithTimeout on the context and pass that from the plugin managers Stop method already?

That's what I had in mind, yes.

>  I considered something like that but would that not mean that if any plugin called cancel on that context it would mean all configured plugins would consider the deadline to be exceeded?

No, not really. If I understand you correct, this example should point at the issue (or non-issue, for that matter):
```go
ctx0, cancel := context.WithCancel(ctx)
go fun1(ctx)
cancel()
// here, ctx0 has been cancelled, ctx has NOT
```

Reply:
Thanks @srenatus - I'll try that 👍

Reply:
@srenatus - Check the [last commit](https://github.com/open-policy-agent/opa/pull/2746/commits/641fc1e0bb8c677fd8dc7fd36f0a0063e078e860) for my changes on this. Looks a lot better, though there are still some things of concern which we might want to look into, such as the plugins getting `Stop`ed in sequence.. this would allow a singe plugin to hijack the whole graceful shutdown period for itself.. I'm not sure if there are implications to calling Stop on each in parallel though.